<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>3D Point Cloud Object Detection - Inference</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_main">

    <!-- Backbone Selector -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_backbone">
      <item>
       <widget class="QLabel" name="backboneLabel">
        <property name="text">
         <string>Backbone Architecture:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="backboneComboBox">
        <item>
         <property name="text">
          <string>PointPillar</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>PV-RCNN</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Model Checkpoint (Moved under Backbone Selector) -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_modelCheckpoint">
      <item>
       <widget class="QLabel" name="modelCheckpointLabel">
        <property name="text">
         <string>Model Checkpoint:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="modelCheckpointLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="browseModelCheckpointButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Inference File/Directory (Edited label) -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_inferenceFile">
      <item>
       <widget class="QLabel" name="inferenceFileLabel">
        <property name="text">
         <string>Inference File/Directory:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="inferenceFileLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="browseInferenceFileButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Checkbox for 2D or 3D -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2D3DCheckBox">
      <item>
       <widget class="QCheckBox" name="is2DCheckBox">
        <property name="text">
         <string>2D Model Inference</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_5">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
       </spacer>
      </item>
     </layout>
    </item>

    <!-- Truth Label Directory -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_truthLabelDirectory">
      <item>
       <widget class="QLabel" name="truthLabelDirectoryLabel">
        <property name="text">
         <string>Truth Label Directory:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="truthLabelDirectoryLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="browseTruthLabelDirectoryButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Save Prediction Labels -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_savePredictions">
      <item>
       <widget class="QLabel" name="savePredictionsLabel">
        <property name="text">
         <string>Save Prediction Labels:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="savePredictionsLineEdit"/>
      </item>
      <item>
       <widget class="QPushButton" name="browseSavePredictionsButton">
        <property name="text">
         <string>Browse</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Progress Bar -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_progressBar">
      <item>
       <widget class="QProgressBar" name="progressBar"/>
      </item>
     </layout>
    </item>

    <!-- Buttons at the bottom -->
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_buttons">
      <item>
       <spacer name="horizontalSpacer_3">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="startInferenceButton">
        <property name="text">
         <string>Start Model Inference</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="startTestingButton">
        <property name="text">
         <string>Start Model Testing</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="text">
         <string>Reset</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cancelButton">
        <property name="text">
         <string>Cancel</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_4">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
